tools/checkpoint_conversion/convert_llama_checkpoints.py

```diff
@@ -0,0 +1,142 @@
+cfg["hidden_dim"] = hf_config["hidden_size"]
+cfg["intermediate_dim"] = hf_config["intermediate_size"]
+cfg["max_sequence_length"] = hf_config["max_position_embeddings"]
+cfg["rope_scaling_type"] = hf_config["rope_scaling"]
+cfg["layer_norm_epsilon"] = hf_config["rms_norm_eps"]
+cfg["num_key_value_heads"] = hf_config["num_key_value_heads"]
+
+
+keras_model = LlamaBackbone(**cfg)
+
+
+keras_model.get_layer("token_embedding").embeddings.assign(
+    hf_wts["embed_tokens.weight"]
+)
+
+for ilayer in range(cfg["num_layers"]):
+    # attention layer
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._self_attention_layer._query_dense.kernel.assign(
+        hf_wts[f"layers.{ilayer}.self_attn.q_proj.weight"]
+        .numpy()
+        .T.reshape((cfg["hidden_dim"], cfg["num_heads"], -1))
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._self_attention_layer._key_dense.kernel.assign(
+        hf_wts[f"layers.{ilayer}.self_attn.k_proj.weight"]
+        .numpy()
+        .T.reshape((cfg["hidden_dim"], cfg["num_key_value_heads"], -1))
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._self_attention_layer._value_dense.kernel.assign(
+        hf_wts[f"layers.{ilayer}.self_attn.v_proj.weight"]
+        .numpy()
+        .T.reshape((cfg["hidden_dim"], cfg["num_key_value_heads"], -1))
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._self_attention_layer._output_dense.kernel.assign(
+        hf_wts[f"layers.{ilayer}.self_attn.o_proj.weight"].numpy().T
+    )
+
+    # MLP
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._feedforward_intermediate_dense.kernel.assign(
+        hf_wts[f"layers.{ilayer}.mlp.up_proj.weight"].numpy().T
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._feedforward_gate_dense.kernel.assign(
+        hf_wts[f"layers.{ilayer}.mlp.gate_proj.weight"].numpy().T
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._feedforward_output_dense.kernel.assign(
+        hf_wts[f"layers.{ilayer}.mlp.down_proj.weight"].numpy().T
+    )
+
+    # LAYERNORM
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._self_attention_layernorm.weight.assign(
+        hf_wts[f"layers.{ilayer}.input_layernorm.weight"]
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._feedforward_layernorm.weight.assign(
+        hf_wts[f"layers.{ilayer}.post_attention_layernorm.weight"]
+    )
+
+
+keras_model.get_layer("layer_norm").gamma.assign(hf_wts["norm.weight"])
+
+token_ids = [1, 2181, 8522, 338]
+padding_mask = [1, 1, 1, 1]
+
+keras_inputs = {
+    "token_ids": torch.tensor([token_ids]),
+    "padding_mask": torch.tensor([padding_mask]),
+}
+
+with torch.no_grad():
+    keras_outputs = keras_model(keras_inputs)
+print("Keras output = ", keras_outputs.numpy())
```

Comment:
Can we add a line that also runs output through the hf version and compares the difference? How close do we get?